internal/runners/branch/add.go

```diff
@@ -0,0 +1,59 @@
+package branch
+
+import (
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+type Add struct {
+	out     output.Outputer
+	project *project.Project
+}
+
+type AddParams struct {
+	Label string
+}
+
+func NewAdd(prime primeable) *Add {
+	return &Add{
+		out:     prime.Output(),
+		project: prime.Project(),
+	}
+}
+
+func (a *Add) Run(params AddParams) error {
+	logging.Debug("ExecuteAdd")
+
+	project, err := model.FetchProjectByName(a.project.Owner(), a.project.Name())
+	if err != nil {
+		return locale.WrapError(err, "branch_list_proejct_err", "Could not get project details for project: {{.V0}}", a.project.Namespace().String())
+	}
+
+	branchID, err := model.AddBranch(project.ProjectID, params.Label)
+	if err != nil {
+		return locale.WrapError(err, "err_add_branch", "Could not add branch")
+	}
+
+	localBranch := a.project.BranchName()
+	branch, err := model.BranchForProjectByName(project, localBranch)
+	if err != nil {
+		return locale.WrapError(err, "err_add_branch_no_default", "Could not retrieve branch information for: {{.V0}}", localBranch)
+	}
+
+	err = model.UpdateBranchTracking(*branchID, a.project.CommitUUID(), branch.BranchID, model.TrackingIgnore)
+	if err != nil {
+		logging.Debug("Unable to update tracking information, attempting to delete branch")
+		derr := model.DeleteBranch(*branchID)
+		if err != nil {
+			logging.Debug("Could not delete branch %s, got error: %v", params.Label, derr)
+		}
```

Comment:
```suggestion
		derr := model.DeleteBranch(*branchID)
		if err != nil {
			logging.Debug("Could not delete branch %s, got error: %v", params.Label, derr)
		}
```

Reply:
Github won't let me commit this one as there are no changes

Reply:
Just want to remove the whole DeleteBranch logic. Reaching this code is unlikely and means something is broken, don't want to start maintaining code to deal with broken API's or something.

Reply:
Makes sense. Do we have a story for `state branch delete`? If there is an issue with `state branch add` at this point a user may not be able to use the same branch label unless they can delete it. If there is no story I can create one.

Reply:
Created it just now: https://www.pivotaltracker.com/story/show/176926065